spec/models/grda_warehouse/enrollment_change_history_spec.rb

```diff
@@ -0,0 +1,57 @@
+
+require 'rails_helper'
+
+RSpec.describe GrdaWarehouse::EnrollmentChangeHistory, type: :model do
+  describe '.create_for_clients_on_date!' do
+    let(:today ) { Date.current }
+    let!(:data_source) { create :grda_warehouse_data_source, authoritative: true}
+    let!(:wh_data_source) { create :grda_warehouse_data_source, source_type: nil, authoritative: false}
+    let!(:organization) { create :hud_organization, data_source: data_source }
+    let!(:project) do
+      create(
+        :grda_warehouse_hud_project,
+        organization: organization,
+        project_type: 4,
+        data_source: data_source,
+      )
+    end
+
+    let(:days_homeless) {30}
+    let!(:source_clients) do
+      3.times.map do
+        source_client = create :hud_client, data_source_id: data_source.id
+        enrollment = create :hud_enrollment, client: source_client, data_source: data_source, project: project, entry_date: today - days_homeless.days
+        client = create(:hud_client, data_source: wh_data_source)
+        create :warehouse_client, source_id: source_client.id, destination_id: client.id, data_source: data_source
+        source_client
+      end
+    end
+    let(:destination_clients) do
+      source_clients.map(&:destination_client)
+    end
+
+    it 'passes a smoke test' do
+      GrdaWarehouse::Tasks::ServiceHistory::Enrollment.all.each(&:rebuild_service_history!)
+
+      expect {
+        described_class.create_for_clients_on_date!(client_ids: destination_clients.map(&:id), date: today )
+      }.to change(described_class, :count).by(3).
+      and make_database_queries(count: 60)
```

Comment:
60 queries for 3 client is not ideal

Reply:
This should be roughly the same as the residential enrollments roll-up on the client dashboard.  Perhaps we can focus some effort there in the future and streamline both.